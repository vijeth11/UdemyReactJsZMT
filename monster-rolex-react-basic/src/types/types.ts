export type Monster = {
    id: string;
    name: string;
    email:string;
  }

export type CardProps = {
    monsters: Monster[]
};