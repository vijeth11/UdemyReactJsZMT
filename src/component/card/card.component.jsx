import React from 'react';
import './card.styles.css';

export const Card = (prop) => {
    return (
        <div className='card-container'>
            <img alt="monster" src={`https://robohash.org/${prop.monster.id}?set=set2&size=180x180`}/>
            <h1>{ prop.monster.name }</h1>
            <p>{ prop.monster.email }</p>
        </div>
    );
}