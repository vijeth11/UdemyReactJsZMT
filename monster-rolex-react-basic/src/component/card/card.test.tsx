import {shallow } from 'enzyme';
import { Monster } from '../../types/types';
import { Card } from './card.component';
let mockData:Monster = {
    email:"test@test.com",
    id:"1",
    name:"test"
}

it("expect to render a card component",() => {
expect(shallow(<Card monster={mockData}/>).length).toEqual(1);
expect(shallow(<Card monster={mockData}/>)).toMatchSnapshot();
});