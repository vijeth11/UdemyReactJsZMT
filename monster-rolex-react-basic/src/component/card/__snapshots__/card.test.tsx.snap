// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render a card component 1`] = `
<div
  className="card-container"
>
  <img
    alt="monster"
    src="https://robohash.org/1?set=set2&size=180x180"
  />
  <h1>
    test
  </h1>
  <p>
    test@test.com
  </p>
</div>
`;
